lib/api.js

```diff
@@ -194,13 +194,28 @@ var api = {
     reqOptions.qs = options.qs;
     reqOptions.headers = options.headers;
 
+    var requestFunction = function() {
+      return _request(reqOptions);
+    };
     if (options.auth === true) {
-      return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
-        return _request(reqOptionsWithToken);
-      });
+      requestFunction = function() {
+        return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+          return _request(reqOptionsWithToken);
+        });
+      };
     }
 
-    return _request(reqOptions);
+    if (!options.retryCodes) {
```

Comment:
Might be more concise/readable to do:

```js
return requestFunction().catch(function(err) {
  if (options.retryCodes && _.includes(...)) {
    // ...
  }
  return Promise.reject(err);
});
```

Not a big deal tho.

Reply:
May also not need to do the `options.retryCodes` part if lodash includes guards against null (which it probably does).